### PR TITLE
Repeat feature UI changes

### PR DESCRIPTION
**Pull request by vermayash8**

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java [hunk 1]
```diff
@@ -322,8 +346,12 @@ private void resetTask(TaskModel task) {
...
-        int state = TaskStateUtil.getTaskState(this, task);
+        // Note: This has been added to reset the repeatable reminders that have been marked done
+        // by mistake. It is assumed that we don't need to set the nextStartDate back to what it
+        // was in case of EXPIRED.
+        task.setNextStartDate(LocalDate.fromDateFields(new Date()));
 
+        int state = TaskStateUtil.getTaskState(this, task);
         if (state == TaskStateUtil.STATE_EXPIRED) {
             task.setIsDone(1);
             task.setSnoozedAt(taskSnoozedAt);
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java]: In the else part, why are we setting lastTriggered = null?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java [hunk 2]
```diff
@@ -341,15 +369,14 @@ private void resetTask(TaskModel task) {
      * Sets the task as done and sets the Done button.
      */
     private void markTaskAsDone(TaskModel task) {
-        task.setIsDone(1);
-        mTaskRepository.updateTask(task);
+        TaskActionUtils.onTaskMarkedDone(getApplicationContext(), task);
         setDoneButton(task);
+        // TODO (minor): should we show the same message to both repeatable and non-repeatable?
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java]: I think we should. Remove TODO?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java [hunk 3]
```diff
@@ -270,9 +276,8 @@ private void fillDataForEditing(final TaskModel task) {
         endDateTv.setTag(task.getEndDate());
 
         // Repeat options.
-        String[] repeatOptions = getResources().getStringArray(R.array.creator_repeat_options);
-        repeatTv.setTag(task.getRepeatType());
-        repeatTv.setText(repeatOptions[task.getRepeatType()]);
+        repeatSwitch.setChecked(task.getRepeatType() == DbConstants.REPEAT_DAILY);
+        // TODO: Setup weekday options while editing(if needed). >>>
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: TO be done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java [hunk 4]
```diff
@@ -569,39 +578,49 @@ private void saveTask() {
...
+                    @Override
+                    public void onWeekdaysItemClicked(int i, WeekdaysDataItem weekdaysDataItem) {
+                        // Since our repeat codes require Monday as 1 and sunday as 7, we'll need
+                        // to change to that format from calendar day id.
+                        int calendarDayId = weekdaysDataItem.getCalendarDayId();
+                        int ourDayIndex = 7;
+                        if (calendarDayId != Calendar.SUNDAY) {
+                            ourDayIndex = calendarDayId - 1;
+                        }
+                        int dayCode = 1 << (ourDayIndex - 1);
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: Shouldn't we use the constants here? This requires calculating and shifting every time.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java [hunk 5]
```diff
@@ -236,8 +247,21 @@ private void showNote(TaskModel task) {
...
+                    && task.getStartDate().compareTo(today) <= 0) {
+                Log.i(TAG, "Task has been set as done.");
+                doneButton.setText(getString(R.string.detail_button_reset));
+            } else {
+                Log.i(TAG, "Task has been set as not done.");
+                doneButton.setText(R.string.detail_button_mark_done);
+            }
         } else {
             Log.i(TAG, "Task has been set as done.");
             doneButton.setText(R.string.detail_button_reset);
```
[the remark this answers is not in the document]
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java]: Yes,  we're setting it to today while resetting.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java (hunk 1 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java]: I've added a note and we'll do this when we'll start showing last Triggered time.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java (hunk 2 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/DetailActivity.java]: Removed. What about the message "Task has been marked done for the day."

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java (hunk 4 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: Done.

**Pull request merged**